sdk/trace/batch_span_processor.go

```diff
@@ -223,11 +230,15 @@ func (bsp *BatchSpanProcessor) processQueue() {
 // drainQueue awaits the any caller that had added to bsp.stopWait
 // to finish the enqueue, then exports the final batch.
 func (bsp *BatchSpanProcessor) drainQueue() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
If `bsp.exportSpans` doesn't spawn goroutines, this won't have any effect. The calls to `bsp.exportSpans` would always return _before_ we'd cancel the `context.Context` we're handing to them.

Reply:
Sure, but as I mentioned in [my other comment](https://github.com/open-telemetry/opentelemetry-go/pull/1608#discussion_r583727120):

> I've added a cancel function to be called ... to be sure if the downstream export is incorrectly asynchronous it will be canceled.

I'm not sure I fully understood [your previous comment](https://github.com/open-telemetry/opentelemetry-go/pull/1608#discussion_r583640780) based on this. I had the same initial reaction to your comment, that `bsp.exportSpans` doesn't spawn goroutines and both `drainQueue` and `processQueue` are only called from a single location with the only interrupt signal being the stopCh which is already handled. Did I miss something?

Reply:
My original comment wasn't suggesting that we need to cancel these (background) `Context`s within these methods, but rather noting that it looks suspicious that we're creating background `Context`s at all. It's a sign that something is missing here, probably due to a backward compatibility requirement. If we need a `Context` but we can't ask for one from our callers, then we're stuck faking it when we call on other functions that accept a `Context`.

It looks like a case for `context.TODO`, even though that doesn't change the behavior at all. It's just a way of admitting that we're stuck calling on things without proper arguments available.

Reply:
Gotcha. I'm not sure the user will ever be able to pass a context for this operation given it is asynchronous and a context is used to handle synchronous processes.

I can revert to remove the cancel function, but I'm not sure setting to a TODO context is appropriate if we never plan to support user context here.

Reply:
For reference, this is where this and the processQueue are kicked off:

https://github.com/open-telemetry/opentelemetry-go/blob/992837f195eacbd84f0596bfaa79470dba7408e5/sdk/trace/batch_span_processor.go#L105-L109

Reply:
The recently active golang/go#44143 comes to mind. Your comment here echoes Russ Cox's https://github.com/golang/go/issues/44143#issuecomment-776938138. In the cases discussed there, we have a sibling function that _does_ take a `Context`, and the one that doesn't fakes it by calling on the other than does. Here, we lack that proper sibling function.

If we have no `Context` available for callers to control, we never will, and we just intend for this cancel-aware procedure to run for as long as it takes, then I suppose `context.Background` is the right choice.

Reply:
> The recently active [golang/go#44143](https://github.com/golang/go/issues/44143) comes to mind. Your comment here echoes Russ Cox's [golang/go#44143 (comment)](https://github.com/golang/go/issues/44143#issuecomment-776938138).

That is an interesting read, and yeah, it seems like Russ Cox's sentiment is similar to what I was talking about.

> If we have no `Context` available for callers to control, we never will, and we just intend for this cancel-aware procedure to run for as long as it takes, then I suppose `context.Background` is the right choice.

:+1: 

Though, working on this change makes me realize this span processor could potentially be refactored to use context to handle process flow and timeouts instead of multiple channels for different types of signals. Given this logic can be refactored after an RC though, I'll leave that for the future.